### PR TITLE
secrets: add ErrorAs function to Keeper

### PR DESCRIPTION
**Pull request by shantuo**
Fixes: #1261

### CODE REVIEW

File: secrets/awskms/kms.go [hunk 1]
```diff
@@ -14,12 +14,19 @@
...
+// awskms exposes the following type for As:
+//  - Error: awserr.Error
 package awskms // import "gocloud.dev/secrets/awskms"
 
 import (
 	"context"
 	"errors"
 
+	"github.com/aws/aws-sdk-go/aws/awserr"
+
```
Comment (vangent) [inline secrets/awskms/kms.go]: Nit: no need for blank line here.

File: secrets/drivertest/drivertest.go [hunk 2]
```diff
@@ -54,6 +64,16 @@ func RunConformanceTests(t *testing.T, newHarness HarnessMaker) {
 	t.Run("TestDecryptMalformedError", func(t *testing.T) {
 		testDecryptMalformedError(t, newHarness)
 	})
+	t.Run("TestAs", func(t *testing.T) {
```
Comment (vangent) [inline secrets/drivertest/drivertest.go]: You can also add a conformance `AsTest` that verifies `As` fails when passed `nil`:

https://github.com/google/go-cloud/blob/master/runtimevar/drivertest/drivertest.go#L72-L93
https://github.com/google/go-cloud/blob/master/runtimevar/drivertest/drivertest.go#L119-#L119

File: secrets/example_test.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"gocloud.dev/secrets/gcpkms"
+	"google.golang.org/grpc/status"
+)
+
+func ExampleKeeper_ErrorAs() {
```
Comment (vangent) [inline secrets/example_test.go]: We need an example for using `secrets` overall. I don't think any of our packages have examples for using `As`.

I suggest:

* Dropping this file from the PR.
* Send a separate PR adding this file, but make the example use `localsecrets` and actually work.

I filed a separate PR (#1274) to improve the documentation/examples for `As`, so we can handle the `As` example there.

File: secrets/localsecrets/localsecrets.go [hunk 4]
```diff
@@ -78,5 +78,10 @@ func (k *keeper) Decrypt(ctx context.Context, message []byte) ([]byte, error) {
 	return decrypted, nil
 }
 
+// ErrorAs implements driver.Keeper.ErrorAs.
```
Comment (vangent) [inline secrets/localsecrets/localsecrets.go]: Please add an `As` section to the package docstring just saying that it doesn't support any types for `As`.
https://github.com/google/go-cloud/blob/master/blob/memblob/memblob.go#L27-L29

File: secrets/vault/vault.go [hunk 5]
```diff
@@ -96,6 +96,11 @@ func (k *keeper) Encrypt(ctx context.Context, plaintext []byte) ([]byte, error)
 	return []byte(secret.Data["ciphertext"].(string)), nil
 }
 
+// ErrorAs implements driver.Keeper.ErrorAs.
```
Comment (vangent) [inline secrets/vault/vault.go]: Same here, add to package docstring even if it doesn't support any types.

What error type does `Vault` return? Maybe add to the `TODO` below, something like "classify vault error codes and add a type for ErrorAs".

File: secrets/drivertest/drivertest.go (hunk 2 above)
Comment (shantuo) [reply secrets/drivertest/drivertest.go]: Done.

File: secrets/example_test.go (hunk 3 above)
Comment (shantuo) [reply secrets/example_test.go]: Done.

File: secrets/localsecrets/localsecrets.go (hunk 4 above)
Comment (shantuo) [reply secrets/localsecrets/localsecrets.go]: Done.

File: secrets/vault/vault.go (hunk 5 above)
Comment (shantuo) [reply secrets/vault/vault.go]: Done.

**Comment by jba** [comment]
LGTM.
